var/spack/repos/builtin/packages/py-python-logstash/package.py

```diff
@@ -0,0 +1,42 @@
+##############################################################################
+# Copyright (c) 2013-2018, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/spack/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+
+
+class PyPythonLogstash(PythonPackage):
+    """Python logging handler for Logstash."""
+
+    homepage = "https://github.com/vklochan/python-logstash"
+    url      = "https://files.pythonhosted.org/packages/4e/8d/7ff2e8e8e2613e7bb7654790480bb4cf51a55721371adbb631b16cb16dce/python-logstash-0.4.6.tar.gz"
+
+    version('0.4.6', '26fafa0ea306025fb7644d70cb38982a')
+
+    def patch(self):
+        filter_file(r'from distutils.core import setup', 
+                    'from setuptools import setup', 
+                    'setup.py')
```

Comment:
Can you explain this patch?

Reply:
@adamjstewart Spack PythonPackage packages use py-setuptools for build/install. py-python-logstash is still using distutils (which is deprecated). This was causing the build to fail when it couldn't find distutils. I thought it made more sense to patch py-python-logstash to use setuptools than to create a package for a deprecated packages and figure out how to substitute that into the python build system class.

Reply:
`PythonPackage` doesn't use `py-setuptools`. I don't think `distutils` is deprecated. I'm pretty sure `distutils` is built into Python, so I don't see how it could not be found. There are currently 99 Python packages in Spack that don't use setuptools:
```console
$ grep -L setuptools $(grep -l PythonPackage */package.py) | wc -l
      99
```